eio/csv_eio.mli

```diff
@@ -0,0 +1,131 @@
+(* File: csv_eio.mli
+
+   Copyright (C) 2023-
+
+     Christophe Troestler <Christophe.Troestler@umons.ac.be>
+     WWW: http://math.umons.ac.be/an/software/
+
+   This library is free software; you can redistribute it and/or modify
+   it under the terms of the GNU Lesser General Public License version 2.1 or
+   later as published by the Free Software Foundation, with the special
+   exception on linking described in the file LICENSE.
+
+   This library is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+   LICENSE for more details. *)
+
+(** Lwt interface to the CSV library.
+
+   This module only offers Lwt input/output functions for CSV files.
+   {!Csv} provides additional functions to transform CSV data. *)
+
+type t = Csv.t
+
+
+(** {2 Input} *)
+
+type in_channel
+(** Stateful handle to input CSV files. *)
+
+val of_source : ?separator:char -> ?strip: bool ->
+                 ?has_header: bool -> ?header: string list ->
+                 ?backslash_escape: bool -> ?excel_tricks:bool ->
+                 ?fix:bool ->
+                 #Eio.Flow.source -> in_channel
```

Comment:
Eio uses objects for IO, they explain their reasoning [here](https://github.com/ocaml-multicore/eio/blob/a6c0be9affd1793613e5e23bba4258c9f56d2c36/doc/rationale.md#dynamic-dispatch).

`#Eio.Flow.source` accepts any kind of readable input.